espree.js

```diff
@@ -799,107 +800,120 @@ function scanNumericLiteral() {
     };
 }
 
-// 7.8.4 String Literals
+/**
+ * Scan a string escape sequence and return its special character.
+ * @param {string} ch The starting character of the given sequence.
+ * @returns {Object} An object containing the character and a flag
+ * if the escape sequence was an octal.
+ * @private
+ */
+function scanEscapeSequence(ch) {
+    var code,
+        unescaped,
+        restore,
+        escapedCh,
+        octal = false;
+
+    // An escape sequence cannot be empty
+    if (!ch) {
+        throwError({}, Messages.UnexpectedToken, "ILLEGAL");
```

Comment:
The old code did this:          
```
} else if (ch === "\\") {
  ch = source[index++];
  if (!ch || !syntax.isLineTerminator(ch.charCodeAt(0))) {
    // parse escape sequence
  } else {
    ++lineNumber;
     if (ch === "\r" && source[index] === "\n") {
      ++index;
    }
    lineStart = index;
  }
```

Since index keeps getting incremented, its hard to see what's being checked and what isn't. It looks like `!ch` was meant to be checked but instead it was just being treated like a line terminator.